lib/ecto/migration/runner.ex

```diff
@@ -16,7 +17,12 @@ defmodule Ecto.Migration.Runner do
   """
   def run(repo, module, direction, operation, migrator_direction, opts) do
     level = Keyword.get(opts, :log, :info)
-    start_link(repo, direction, migrator_direction, level)
+    ensure_manager_started()
+    Manager.put_migration(self(), Keyword.get(opts, :prefix, nil))
```

Comment:
I think the prefix should be stored in the runner. :) All the agent should really do is to return the runner pid from the current migration process.